The job **`tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/18112868831/job/51542744094) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/51542744094) [(plain)](https://github.com/rust-lang/rust/commit/11f8023b479c6d12706221a7470ce5a927fb2fd1/checks/51542744094/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling termcolor v1.4.1
error: unused import: `DocTests`
##[error]  --> src/bootstrap/src/core/builder/cargo.rs:13:48
   |
13 |     BootstrapCommand, CLang, Compiler, Config, DocTests, DryRun, EXTRA_CHECK_CFGS, GitRepo, Mode,
   |                                                ^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(unused_imports)]`

error: could not compile `bootstrap` (lib) due to 1 previous error
failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --jobs=default --manifest-path /checkout/src/bootstrap/Cargo.toml -Zroot-dir=/checkout --locked --features build-metrics
Build completed unsuccessfully in 0:00:28
make: *** [Makefile:99: prepare] Error 1
Command failed. Attempt 2/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (/checkout/src/bootstrap)
error: unused import: `DocTests`
##[error]  --> src/bootstrap/src/core/builder/cargo.rs:13:48
   |
13 |     BootstrapCommand, CLang, Compiler, Config, DocTests, DryRun, EXTRA_CHECK_CFGS, GitRepo, Mode,
   |                                                ^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(unused_imports)]`

error: could not compile `bootstrap` (lib) due to 1 previous error
failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --jobs=default --manifest-path /checkout/src/bootstrap/Cargo.toml -Zroot-dir=/checkout --locked --features build-metrics
Build completed unsuccessfully in 0:00:02
make: *** [Makefile:99: prepare] Error 1
Command failed. Attempt 3/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (/checkout/src/bootstrap)
error: unused import: `DocTests`
##[error]  --> src/bootstrap/src/core/builder/cargo.rs:13:48
   |
13 |     BootstrapCommand, CLang, Compiler, Config, DocTests, DryRun, EXTRA_CHECK_CFGS, GitRepo, Mode,
   |                                                ^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(unused_imports)]`

error: could not compile `bootstrap` (lib) due to 1 previous error
failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --jobs=default --manifest-path /checkout/src/bootstrap/Cargo.toml -Zroot-dir=/checkout --locked --features build-metrics
Build completed unsuccessfully in 0:00:02
make: *** [Makefile:99: prepare] Error 1
Command failed. Attempt 4/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (/checkout/src/bootstrap)
error: unused import: `DocTests`
##[error]  --> src/bootstrap/src/core/builder/cargo.rs:13:48
   |
13 |     BootstrapCommand, CLang, Compiler, Config, DocTests, DryRun, EXTRA_CHECK_CFGS, GitRepo, Mode,
   |                                                ^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(unused_imports)]`

error: could not compile `bootstrap` (lib) due to 1 previous error
failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --jobs=default --manifest-path /checkout/src/bootstrap/Cargo.toml -Zroot-dir=/checkout --locked --features build-metrics
Build completed unsuccessfully in 0:00:02
make: *** [Makefile:99: prepare] Error 1
Command failed. Attempt 5/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (/checkout/src/bootstrap)
error: unused import: `DocTests`
##[error]  --> src/bootstrap/src/core/builder/cargo.rs:13:48
   |
13 |     BootstrapCommand, CLang, Compiler, Config, DocTests, DryRun, EXTRA_CHECK_CFGS, GitRepo, Mode,
   |                                                ^^^^^^^^
   |
   = note: `-D unused-imports` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(unused_imports)]`

error: could not compile `bootstrap` (lib) due to 1 previous error
failed to run: /checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo build --jobs=default --manifest-path /checkout/src/bootstrap/Cargo.toml -Zroot-dir=/checkout --locked --features build-metrics
Build completed unsuccessfully in 0:00:02
make: *** [Makefile:99: prepare] Error 1
The command has failed after 5 attempts.
  local time: Mon Sep 29 22:50:21 UTC 2025
  network time: Mon, 29 Sep 2025 22:50:21 GMT
```

</details>

--- a/src/bootstrap/src/core/builder/cargo.rs
+++ b/src/bootstrap/src/core/builder/cargo.rs
@@ -10,7 +10,7 @@ use crate::core::config::flags::Color;
 use crate::utils::build_stamp;
 use crate::utils::helpers::{self, LldThreads, check_cfg_arg, linker_args, linker_flags};
 use crate::{
-    BootstrapCommand, CLang, Compiler, Config, DocTests, DryRun, EXTRA_CHECK_CFGS, GitRepo, Mode,
+    BootstrapCommand, CLang, Compiler, Config, DryRun, EXTRA_CHECK_CFGS, GitRepo, Mode,
     RemapScheme, TargetSelection, command, prepare_behaviour_dump_dir, t,
 };
 
@@ -851,8 +851,6 @@ impl Builder<'_> {
 
         rustflags.arg("-Zmacro-backtrace");
 
-        let want_rustdoc = self.doc_tests != DocTests::No;
-
         // Clear the output directory if the real rustc we're using has changed;
         // Cargo cannot detect this as it thinks rustc is bootstrap/debug/rustc.
         //
@@ -881,7 +879,8 @@ impl Builder<'_> {
             .env("RUSTC_REAL", self.rustc(compiler))
             .env("RUSTC_STAGE", build_compiler_stage.to_string())
             .env("RUSTC_SYSROOT", sysroot)
-            .env("RUSTC_LIBDIR", libdir)
+            .env("RUSTC_LIBDIR", &libdir)
+            .env("RUSTDOC_LIBDIR", libdir)
             .env("RUSTDOC", self.bootstrap_out.join("rustdoc"))
             .env("RUSTDOC_REAL", rustdoc_path)
             .env("RUSTC_ERROR_METADATA_DST", self.extended_error_dir());
@@ -917,11 +916,6 @@ impl Builder<'_> {
 
         if let Some(stack_protector) = &self.config.rust_stack_protector {
             rustflags.arg(&format!("-Zstack-protector={stack_protector}"));
-        }
-
-        if !matches!(cmd_kind, Kind::Build | Kind::Check | Kind::Clippy | Kind::Fix) && want_rustdoc
-        {
-            cargo.env("RUSTDOC_LIBDIR", self.rustc_libdir(compiler));
         }
 
         let debuginfo_level = match mode {

